### ISSUE TITLE
Mastodon -> Bsky crossposted images failed to load

### ISSUE THREAD

**Issue by rowantobias** [snarfed/bridgy-fed#2451]
I've been having consistent issues with crossposted art posts (from my mastodon -> bsky) lacking visuals. Alt text federates just fine, but the images themselves fail to load.

You can see an example of this in individual posts ([post 1](https://bsky.app/profile/swiftyshq.critter.cafe.ap.brid.gy/post/3mjko4ylcklj2), [post 2](https://bsky.app/profile/swiftyshq.critter.cafe.ap.brid.gy/post/3mj5zddtif6n2)), or just by looking at my bridged profile [here](https://bsky.app/profile/swiftyshq.critter.cafe.ap.brid.gy).

I'm not sure how long this issue has been present. Reposts from my account are fine (ie, if I repost someone else's post), but original posts I make do not federate their images correctly. I've had a few people reach out to tell me it's borked, but I'm not sure there's anything I can do to fix it.

Thanks for your time!

**Comment by snarfed** [comment]
Hey, sorry for the trouble! Sounds like https://fed.brid.gy/docs#bluesky-image-cdn ?

**Comment by rowantobias** [comment]
seems like! per my instance admin:
> it's bluesky's own content delivery network that doesn't have a user agent. 
> there's an [issue open about this [github.com]](https://github.com/bluesky-social/atproto/issues/3504) that's now been open for over a year, it doesn't seem like bluesky intends to fix this.
> on bridgy's end, the issue was "resolved" with a "fix" that is essentially a bypass of the bot protection we have in place that could be abused by anyone.
 >in short, the part of bluesky that grabs images refuses to identify itself to the servers it's connecting to and gets blocked as a result, and bluesky doesn't seem to care. we can't fix this on our end without compromising security and exposing ourselves to more scraper DDOS attacks that are only becoming more common, as well as general data scraping. sorry :/

so i guess it's not something that can be fixed without some compromise. ah well. thanks for the help IDIng the issue!

**Comment by snarfed** [comment]
You're welcome! Sorry I don't have better news. And thanks for passing along your admin's notes. I don't really follow "on bridgy's end, the issue was "resolved" with a "fix" that is essentially a bypass of the bot protection"; we don't have a fix for this on our end, and we're not trying to bypass bot protection anywhere afaik...but no matter. 🤷

**Comment by snarfed** [comment]
Tentatively closing. Feel free to reopen!
